spec/features/accompaniment_leader/viewing_and_reporting_on_accompaniments_spec.rb

```diff
@@ -15,18 +14,18 @@
     end
 
     it 'displays full details of accompaniments in the current week' do
-      expect(page).to have_content(current_week_activity.friend.first_name)
-      expect(page).to have_content(current_week_activity.friend.phone)
+      expect(page).to have_content(activity.friend.first_name)
+      expect(page).to have_content(activity.friend.phone)
     end
 
     it 'displays full details of accompaniments in the upcoming week' do
-      expect(page).to have_content(next_week_activity.friend.first_name)
-      expect(page).to have_content(current_week_activity.friend.phone)
+      expect(page).to have_content(activity.friend.first_name)
+      expect(page).to have_content(activity.friend.phone)
```

Comment:
I'm realizing that in the original spec, line 24 is a copy/paste error.  It should be next_week_activity, not current_week_activity (as that is tested on line 19).

I like the direction you are going to get rid of the differentiation between 'current week' and 'next week', which you have started by replacing current_week_activity/next_week_activity with 'activity'.  So we don't have two blocks that are asserting the same thing, we should get rid of one of them and then rename to "it 'displays full details of upcoming accompaniments'".  It looks like you have already done this with the volunteer spec.

Then, and I'll write a separate card for this so it's not something that needs to be changed in this PR, we can add a model spec to cover the 'upcoming_two_weeks' on Activity which, I think, is the best way to test that the method is selecting the right activities.

Reply:
good catch. just missed this in review. i'm noticing some non-deterministic failures in some other specs just now. https://gist.github.com/rsl/0361949dc827f4b3cbe5ea997422abd1 when the first one failed i immediately reran and it gave me the second output, third time ran fine. should i make a card for looking into this?

Reply:
@rsl that would be great!  Yah, I have seen this one flap as well, and adding a card to look into it would be great.  Thank you!